File: src/main/java/net/lifecity/mc/skillmaster/game/GameList.java
package net.lifecity.mc.skillmaster.game;

import net.lifecity.mc.skillmaster.user.SkillUser;

import java.util.ArrayList;

public class GameList extends ArrayList<Game> {

    /**
     * ユーザーが参加しているゲームを取得します
     * @param user このユーザーを使います
     * @return 参加していたらtrue
     */
    public Game getFromUser(SkillUser user) {
        for (Game game : this) {
            if (game.joined(user))
                return game;
        }
        return null;
    }

    /**
     * ユーザーがなんらかのゲームに参加しているかを返します
     * @param user このユーザーを使います
     * @return 参加していたらtrueを返します
     */
    public boolean inGamingUser(SkillUser user) {
        Game game = getFromUser(user);
        return game != null;
    }
}

File: src/main/java/net/lifecity/mc/skillmaster/game/GameList.kt
package net.lifecity.mc.skillmaster.game

import net.lifecity.mc.skillmaster.GameNotFoundException
import net.lifecity.mc.skillmaster.SkillUserNotFoundException
import net.lifecity.mc.skillmaster.user.SkillUser

class GameList {

    val list = mutableListOf<Game>()

    /**
     * ユーザーが参加しているゲームを取得します
     * @param user このユーザーを使います
     * @return 参加していたらtrue
     */
    fun getFromUser(user: SkillUser): Game {
        for (game in list) {
            if (game.gameManager.joined(user)) return game
        }
        throw GameNotFoundException("${user.player.name} do not join any games now")
    }

    /**
     * ユーザーがなんらかのゲームに参加しているかを返します
     * @param user このユーザーを使います
     * @return 参加していたらtrueを返します
     */
    fun inGamingUser(user: SkillUser) : Boolean{
        for (game in list) {
            if (game.gameManager.joined(user)) return true
        }
        return false
    }

}